internal/olm/client/client.go

```diff
@@ -204,6 +206,46 @@ func (c Client) DoCSVWait(ctx context.Context, key types.NamespacedName) error {
 				})
 				return false, nil
 			}
+			// Check deployment status
+			if ctx.Err().Error() == "DeadlineExceeded" {
```

Comment:
If the context deadline has exceeded, `PollImmediateUntil` will return an error that can be unwrapped, so you can run this logic like
```go
err := wait.PollImmediateUntil(time.Second, csvPhaseSucceeded, ctx.Done())
if err != nil && errors.Is(err, context.DeadlineExceeded) {
	var depCheckErr error
	// Logic
	return depCheckErr
}
return err
```